### PR TITLE
Add a doc on side channel countermeasures and known side channel issues

### PR DESCRIPTION
**Pull request by randombit**
@jurajsomorovsky @neusdan @cordney @neverhub review from any/all would be appreciated.

### CODE REVIEW

File: doc/side_channels.rst [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+a single side channel silent algorithm.
+
+See point_gfp.cpp for details.
+
+ECDH
+----------------------
+
+ECDH verifies (through its use of OS2ECP) that all input points
+received from the other party satisfy the curve equation. This
+prevents twist attacks. The same check is performed on the output
```
Comment (jurajsomorovsky) [inline doc/side_channels.rst]: Useful References:

[SafeCurves] Bernstein, Lange: SafeCurves: choosing safe curves for elliptic-curve cryptography. (http://safecurves.cr.yp.to)

File: doc/side_channels.rst [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+useful information for an attacker. Every BOTAN_BLINDING_REINIT_INTERVAL
+(default 32) operations, a new starting point is chosen.
+
+RSA signing uses the CRT optimization, which is much faster but vulnerable to
+trivial fault attacks [RsaFault] which can result in the key being entirely
+compromised. To protect against this (or any other computational error which
+would have the same effect as a fault attack in this case), after every private
+key operation the result is checked for consistency with the public key. This
+introduces only slight additional overhead and blocks most fault attacks; it is
+possible to use a second fault attack to bypass this check but
```
Comment (neusdan) [inline doc/side_channels.rst]: something missing here?

File: doc/side_channels.rst [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+If BOTAN_USE_VOLATILE_MEMSET_FOR_ZERO is set to 0 in build.h (not the default) a
+byte at a time loop through a volatile pointer is used to overwrite the array.
+
+Botan's secure_vector type is a std::vector with a custom allocator. The
+allocator simply calls secure_scrub_memory before freeing memory.
+
+Memory allocation
+----------------------
+
+Some systems including Linux and FreeBSD support applications pinning a small
```
Comment (neusdan) [inline doc/side_channels.rst]: also mention Windows here? VirtualLock / VirtualUnlock

File: doc/side_channels.rst (hunk 2 above)
Comment (randombit) [inline doc/side_channels.rst]: Didn't finish the sentence here, thanks.

File: doc/side_channels.rst (hunk 3 above)
Comment (randombit) [inline doc/side_channels.rst]: Done

**Pull request merged**